src/Microsoft.VisualStudio.Services.Agent/JobNotification.cs

```diff
@@ -55,7 +80,54 @@ public sealed class JobNotification : AgentService, IJobNotification
                 await _outClient.ConnectAsync(cancellationToken);
                 _writeStream = new StreamWriter(_outClient, Encoding.UTF8);
                 _configured = true;
-                Trace.Info("Connection successfull to named pipe {0}", pipeName);
+                Trace.Info("Connection successful to named pipe {0}", pipeName);
+            }
+        }
+
+        public void StartClient(String socketAddress)
```

Comment:
The goal here is that we never fail a job if connecting to the socket fails. That's we set the _isconfigured flag last and make sure that we check it before writing to the socket. If this connection doesn't work, we will at least keep trying to run the job.